### ISSUE TITLE
Partitioning with keys (Group by key) in S3 sink kafka connect

### ISSUE THREAD

**Issue by arezvani** [confluentinc/kafka-connect-storage-cloud#548]
Hi,
Can we partition our output in s3 sink connector with key?

For example if I have these couple of key and value in kafka topic:

key : value
1 : 1
2 : 2
3 : 3
1 : 4 
2 : 5
3 : 6

our output should be like that:

1.json -> 1,4
2.json -> 2,5
3.json -> 3,6

**Comment by OneCricketeer** [comment]
No, you cannot create files like this. The format of the file names are hard coded to topic-partition+offset. 

Instead, you can use FieldPartitioner and have paths by key 

```
key=1/topic+1+0000.json
key=2/topic+1+0000.json
key=3/topic+2+0000.json
```
